graphql/resolvers.go

```diff
@@ -44,6 +44,80 @@ func (r *Resolver) Task() TaskResolver {
 
 type mutationResolver struct{ *Resolver }
 
+func (r *taskResolver) DependsOn(ctx context.Context, t *restModel.APITask) ([]*Dependency, error) {
+	depIds := []string{}
+	for _, dep := range t.DependsOn {
+		depIds = append(depIds, dep.TaskId)
+	}
+
+	dependencyTasks, err := task.Find(task.ByIds(depIds).WithFields(task.DisplayNameKey, task.StatusKey,
+		task.ActivatedKey, task.BuildVariantKey, task.DetailsKey, task.DependsOnKey))
+
+	if err != nil {
+		return nil, InternalServerError.Send(ctx, fmt.Sprintf("Cannot find dependency tasks for task %s:%s", *t.Id, err.Error()))
+	}
+
+	taskMap := map[string]*task.Task{}
+	for i := range dependencyTasks {
+		taskMap[dependencyTasks[i].Id] = &dependencyTasks[i]
+	}
+
+	task, err := task.FindOneId(*t.Id)
+	if err != nil {
+		return nil, ResourceNotFound.Send(ctx, fmt.Sprintf("Unable to find task %s: %s", *t.Id, err.Error()))
+	}
+	state, err := task.BlockedState()
+	if err != nil {
+		return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error getting blocked state for task %s:%s", task.Id, err.Error()))
+	}
+	err = task.CircularDependencies()
+	if err != nil {
+		return nil, InternalServerError.Send(ctx, fmt.Sprintf("Found circular dependency for task %s: %s", task.Id, err.Error()))
+	}
+
+	dependencies := []*Dependency{}
+	for _, dep := range t.DependsOn {
+		task, ok := taskMap[dep.TaskId]
+		if !ok {
+			continue
+		}
+		var metStatus MetStatus
+		if state == evergreen.TaskStatusBlocked {
+			metStatus = "UNMET"
+		} else if task.Status != evergreen.TaskFailed && task.Status != evergreen.TaskSucceeded {
```

Comment:
Instead of rewriting this logic, could you use the task.satisfiesDependency function? You'll need to make it public but I think that's fine